src/devtools/client/debugger/src/components/SourceOutline/getOutlineSymbols.tsx

```diff
@@ -2,19 +2,34 @@ import { fuzzySearch } from "../../utils/function";
 import { groupBy, keyBy } from "lodash";
 import { SourceSymbols, ClassSymbol, FunctionSymbol } from "../../types";
 
-export function getOutlineSymbols(symbols: null | SourceSymbols, filter: string) {
+function addHitCountsToFunctions(functions, hitCounts) {
+  return functions.map(functionSymbol => {
+    const startLine = functionSymbol.location.start.line;
+    const endLine = functionSymbol.location.end.line;
+    for (const line in hitCounts) {
+      if (line >= startLine && line <= endLine) {
+        return { ...functionSymbol, hitCount: hitCounts[line] };
+      }
+    }
+  });
+}
+
+export function getOutlineSymbols(symbols: null | SourceSymbols, filter: string, hitCounts) {
   if (!symbols || symbols.loading) {
     return null;
   }
   const { classes, functions } = symbols;
+  const functionsWithHits = addHitCountsToFunctions(functions, hitCounts);
+  console.log("yo", functionsWithHits);
   const classNames = new Set(classes.map(s => s.name));
-  const functionsByName = keyBy(functions, "name");
-  const filteredFunctions = functions.filter(
+  const functionsByName = keyBy(functionsWithHits, "name");
+  const filteredFunctions = functionsWithHits.filter(
```

Comment:
In the future - I think we'll transition to asking the server for the source's symbols. At that point the server can just return the hitcounts for the symbols as well.